optimum/habana/transformers/generation/stopping_criteria.py

```diff
@@ -14,17 +14,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import time
+from typing import Union
+
 import torch
 
 from optimum.utils import logging
 
 
 logger = logging.get_logger(__name__)
 
+# Instead of returning a tensor describing status of completeness of each sentence
+# we only return a single boolean describing the state of the batch
+# only when needs_tensor_output says so, we return array of booleans
```

Comment:
note: stopping criteria is optimized to return a bool (as opposed to an array of bool) for perf reasons